src/ParquetMsg.chpl

```diff
@@ -122,13 +122,34 @@ module ParquetMsg {
     }
   }
 
-  proc getStrColSize(filename: string, dsetname: string) throws {
-    extern proc c_getStringColumnNumBytes(filename, colname, errMsg): int;
+  proc calcSizesAndOffset(offsets: [] ?t, byteSizes: [] int, filenames: [] string, sizes: [] int, dsetname: string) throws {
+    var (subdoms, length) = getSubdomains(sizes);
+    
+    coforall loc in offsets.targetLocales() do on loc {
+      var locFiles = filenames;
+      var locFiledoms = subdoms;
+      forall (i, filedom, filename) in zip(sizes.domain, locFiledoms, locFiles) {
+        for locdom in offsets.localSubdomains() {
+          const intersection = domain_intersection(locdom, filedom);
+          if intersection.size > 0 {
+            var pqErr = new parquetErrorMsg();
+            var col: [filedom] t;
+            byteSizes[i] = getStrColSize(filename, dsetname, col);
```

Comment:
I forget, what happens if this function throws an error from inside a `forall`? I seem to remember having issues throwing from within a `forall`, but maybe that was with an old version of Chapel.

Reply:
This code is almost identical to the code that was causing the errors in https://github.com/Bears-R-Us/arkouda/issues/1153, so when you are in a `forall`/`coforall`, Chapel will sum up all the errors and return a `TaskErrors` which has all the errors together. The fix that implemented for the bug you were seeing was sort of a workaround for that, and I am not sure why the throwing `TaskErrors` is not caught by the main Arkouda `try`/`catch` in `arkouda_server.chpl`. 

I am planning on investigating that some more in the near feature, but I think you are correct that there may be some issues with throwing in these loops today. Since there are already other instances doing this in the codebase today (specifically https://github.com/Bears-R-Us/arkouda/blob/20f0315bf8f3723f4bc9a8803f1f89cc01d98ae8/src/ParquetMsg.chpl#L100-L123), I was thinking that we should get this in now and then I will hopefully know more about the throwing errors soon, but I have no problem waiting until I've pinned that down if you think that is a safer approach.

Reply:
Thanks for the explanation! I am ok with merging now while the investigation of error handling continues.